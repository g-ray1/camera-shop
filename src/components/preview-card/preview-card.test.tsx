import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import PreviewCard from './preview-card';

const camera = {
  id: 1,
  name: 'Ретрокамера Dus Auge lV',
  vendorCode: 'DA4IU67AD5',
  type: 'Коллекционная',
  category: 'Видеокамера',
  description: 'Немецкий концерн BRW разработал видеокамеру Das Auge IV в начале 80-х годов, однако она до сих пор пользуется популярностью среди коллекционеров и яростных почитателей старинной техники. Вы тоже можете прикоснуться к волшебству аналоговой съёмки, заказав этот чудо-аппарат. Кто знает, может с Das Auge IV начнётся ваш путь к наградам всех престижных кинофестивалей.',
  previewImg: 'img/content/das-auge.jpg',
  level: 'Любительский',
  rating: 4,
  price: 73450,
  previewImg2x: 'img/content/das-auge@2x.jpg',
  previewImgWebp: 'img/content/das-auge.webp',
  previewImgWebp2x: 'img/content/das-auge@2x.webp',
  reviewCount: 17
};

jest.mock('react-redux');

describe('PreviewCard component', () => {
  it('Snapshot', () => {
    const view = render(<PreviewCard product={camera} />, {wrapper: BrowserRouter});

    expect(view).toMatchSnapshot();
  });
});
