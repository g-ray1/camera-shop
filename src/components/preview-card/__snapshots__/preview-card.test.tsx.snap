// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewCard component Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="product-card "
      >
        <div
          class="product-card__img"
        >
          <picture>
            <source
              srcset="/img/content/das-auge.webp, /img/content/das-auge@2x.webp 2x"
              type="image/webp"
            />
            <img
              alt="Ретрокамера Dus Auge lV"
              height="240"
              src="/img/content/das-auge.jpg"
              srcset="/img/content/das-auge@2x.jpg 2x"
              width="280"
            />
          </picture>
        </div>
        <div
          class="product-card__info"
        >
          <div
            class="rate product-card__rate"
          >
            <div
              class="rate product-card__rate"
            >
              <svg
                aria-hidden="true"
                height="16"
                width="17"
              >
                <use
                  xlink:href="#icon-full-star"
                />
              </svg>
              <svg
                aria-hidden="true"
                height="16"
                width="17"
              >
                <use
                  xlink:href="#icon-full-star"
                />
              </svg>
              <svg
                aria-hidden="true"
                height="16"
                width="17"
              >
                <use
                  xlink:href="#icon-full-star"
                />
              </svg>
              <svg
                aria-hidden="true"
                height="16"
                width="17"
              >
                <use
                  xlink:href="#icon-full-star"
                />
              </svg>
              <svg
                aria-hidden="true"
                height="16"
                width="17"
              >
                <use
                  xlink:href="#icon-star"
                />
              </svg>
              <p
                class="visually-hidden"
              >
                Рейтинг:
              </p>
              <p
                class="rate__count"
              >
                <span
                  class="visually-hidden"
                >
                  Всего оценок:
                </span>
                17
              </p>
            </div>
          </div>
          <p
            class="product-card__title"
          >
            Ретрокамера Dus Auge lV
          </p>
          <p
            class="product-card__price"
          >
            <span
              class="visually-hidden"
            >
              Цена:
            </span>
            73450 ₽
          </p>
        </div>
        <div
          class="product-card__buttons"
        >
          <button
            class="btn btn--purple product-card__btn"
            type="button"
          >
            Купить
          </button>
          <a
            class="btn btn--transparent"
            href="/product/1"
          >
            Подробнее
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="product-card "
    >
      <div
        class="product-card__img"
      >
        <picture>
          <source
            srcset="/img/content/das-auge.webp, /img/content/das-auge@2x.webp 2x"
            type="image/webp"
          />
          <img
            alt="Ретрокамера Dus Auge lV"
            height="240"
            src="/img/content/das-auge.jpg"
            srcset="/img/content/das-auge@2x.jpg 2x"
            width="280"
          />
        </picture>
      </div>
      <div
        class="product-card__info"
      >
        <div
          class="rate product-card__rate"
        >
          <div
            class="rate product-card__rate"
          >
            <svg
              aria-hidden="true"
              height="16"
              width="17"
            >
              <use
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="16"
              width="17"
            >
              <use
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="16"
              width="17"
            >
              <use
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="16"
              width="17"
            >
              <use
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="16"
              width="17"
            >
              <use
                xlink:href="#icon-star"
              />
            </svg>
            <p
              class="visually-hidden"
            >
              Рейтинг:
            </p>
            <p
              class="rate__count"
            >
              <span
                class="visually-hidden"
              >
                Всего оценок:
              </span>
              17
            </p>
          </div>
        </div>
        <p
          class="product-card__title"
        >
          Ретрокамера Dus Auge lV
        </p>
        <p
          class="product-card__price"
        >
          <span
            class="visually-hidden"
          >
            Цена:
          </span>
          73450 ₽
        </p>
      </div>
      <div
        class="product-card__buttons"
      >
        <button
          class="btn btn--purple product-card__btn"
          type="button"
        >
          Купить
        </button>
        <a
          class="btn btn--transparent"
          href="/product/1"
        >
          Подробнее
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
